subprojects/packagefiles/openssl/meson.build

```diff
@@ -0,0 +1,452 @@
+project(
+  'openssl',
+  'c',
+  version: '1.1.1k',
+  license : 'Apache-2.0',
+  meson_version: '>= 0.55',
+  default_options: [
+    'warning_level=1',
+    'optimization=3',
+  ],
+)
+
+# Make sure to generate configs in case they are not already
+fs = import('fs')
+if not fs.exists('generated-config')
+  message('Generating OpenSSL configs...')
+  run_command(
+    'generator.sh',
+    check: true,
+    env: ['OPENSSL_VERSION=' + meson.project_version()],
+  )
+endif
+
+include_directories = [
+  'include',
+  'crypto',
+  'crypto/modes',
+  'crypto/ec/curve448',
+  'crypto/ec/curve448/arch_32',
+]
+
+compiler = meson.get_compiler('c')
+
+dependencies = [
+  # TODO: Make this optionally added once we have threading configurable via options
+  dependency('threads'),
+]
+defines = [
+  # Compile out hardware engines.  Most are stubs that dynamically load
+  # the real driver but that poses a security liability when an attacker
+  # is able to create a malicious DLL in one of the default search paths.
+  'OPENSSL_NO_HW',
+]
+c_args = []
+
+asm = 'no-asm'
+asm_opt = get_option('asm')
+
+if asm_opt.disabled()
+  new_gas_or_nasm = false
+  any_gas_or_nasm = false
+else
+  new_gas_or_nasm = (
+    find_program('as', required: false, version: '>=2.26').found() or
+    find_program('nasm', required: false, version: '>=2.11.8').found()
+  )
+  any_gas_or_nasm = (
+    new_gas_or_nasm or
+    find_program('as', required: false).found() or
+    find_program('nasm', required: false).found()
+  )
+endif
+
+if new_gas_or_nasm
+  # Require AVX512IFMA supported. See
+  # https://www.openssl.org/docs/man1.1.1/man3/OPENSSL_ia32cap.html
+  # Currently crypto/poly1305/asm/poly1305-x86_64.pl requires AVX512IFMA.
+  if host_machine.cpu_family() == 'ppc' and host_machine.system() == 'linux'
+    arch_subdir = 'linux-ppc'
+    asm = 'asm'
```

Comment:
nitpicking: instead of repeating `asm = 'asm'` everywhere, you could set it once at the top, then set it to no-asm in the few cases where you have a warning().